src/Core/BasePlugin.php

```diff
@@ -302,4 +310,15 @@ public function middleware(MiddlewareQueue $middlewareQueue): MiddlewareQueue
     public function services(ContainerInterface $container): void
     {
     }
+
+    /**
+     * Register application events.
+     *
+     * @param \Cake\Event\EventManagerInterface $eventsManager The global event manager to register listeners on
+     * @return \Cake\Event\EventManagerInterface
+     */
+    public function events(EventManagerInterface $eventsManager): EventManagerInterface
+    {
+        return $eventsManager;
```

Comment:
It lets you replace the eventmanager if you need to 🤷

Reply:
In a web request the eventsmanager instance gets replaced by the one being returned by this method.
First the plugins are used to add events and then finally the app (so that users can either remove specific plugin listeners or completely replace the instance if they want to)

Also, commands can have their own EventsManager. Thats why I first thought having a separate hook would be better to not mix and match both instances.